src/app/shared/directives/resolve-app-header-display.directive.ts

```diff
@@ -0,0 +1,26 @@
+import { Directive, ElementRef, OnInit, Renderer2 } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { filter } from 'rxjs';
+
+
+@Directive({
+  selector: '[appResolveAppHeaderDisplay]'
+})
+export class ResolveAppHeaderDisplayDirective implements OnInit {
+
+  constructor(private router: Router, private elementRef: ElementRef,
+    private renderer2: Renderer2) { }
+
+  ngOnInit(): void {
+    this.router.events
+      .pipe(filter((event: unknown) => event instanceof NavigationEnd))
+      .subscribe((event) => {
+        if ((event as NavigationEnd)?.url.includes('add')) {
```

Comment:
Use add-record for now since that's the only use case. It can be refactored when the need arises in the future.